file/writable_file_writer.cc

```diff
@@ -275,11 +291,23 @@ IOStatus WritableFileWriter::SyncInternal(bool use_fsync) {
   TEST_SYNC_POINT("WritableFileWriter::SyncInternal:0");
   auto prev_perf_level = GetPerfLevel();
   IOSTATS_CPU_TIMER_GUARD(cpu_write_nanos, env_);
+#ifndef ROCKSDB_LITE
+  FileOperationInfo::TimePoint start_ts;
+  if (ShouldNotifyListeners()) {
+    start_ts = std::chrono::system_clock::now();
+  }
+#endif
   if (use_fsync) {
     s = writable_file_->Fsync(IOOptions(), nullptr);
```

Comment:
Should I treat `Fsync()` and `Sync()` here separately? I didn't because `Sync()` calls `Fsync()` by default underneath.

Reply:
Actually it depends on the underlying FileSystem. In `PosixWritableFile`, `Fsync` calls `fsync()` while `Sync` calls `fdata_sync()`, as in env/io_posix.cc.
I am OK with your current approach, maybe you can add a member in `FileInfo` to denote the type of the operation.
I am also OK if you focus on `Flush` in this PR.

Reply:
Is it better to have just one callback function with different types passed in?